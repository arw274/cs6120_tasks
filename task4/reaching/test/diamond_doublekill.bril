# Block 0 reaching definitions:
# Block 1 (ge) reaching definitions:
#   From block 0 : var b instr 0
#   From block 0 : var cond0 instr 1
# Block 2 (zero) reaching definitions:
#   From block 0 : var b instr 0
#   From block 0 : var cond0 instr 1
# Block 3 (end) reaching definitions: 
#   From block 0 : var cond0 instr 1
#   From block 1 : var b instr 1
#   From block 2 : var b instr 1

@main (input: int) {
    b: int = const one;
    cond0: bool = ge input b;
    br cond0 .ge .zero;
.ge:
    b: int = sub input one;
    jmp .end;
.zero:
    b: int = const input;
    jmp. end;
.end:
    ret;
}