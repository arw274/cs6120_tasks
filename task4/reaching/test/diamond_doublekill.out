Function: main

Block 0: [{'args': ['one'], 'dest': 'b', 'op': 'const', 'type': 'int'}, {'args': ['input', 'b'], 'dest': 'cond0', 'op': 'ge', 'type': 'bool'}, {'args': ['cond0'], 'labels': ['ge', 'zero'], 'op': 'br'}]
Block 1: [{'label': 'ge'}, {'args': ['input', 'one'], 'dest': 'b', 'op': 'sub', 'type': 'int'}, {'labels': ['end'], 'op': 'jmp'}]
Block 2: [{'label': 'zero'}, {'args': ['input'], 'dest': 'b', 'op': 'const', 'type': 'int'}, {'args': ['end'], 'op': 'jmp.'}]
Block 3: [{'label': 'end'}, {'op': 'ret'}]
Block 0 reaching definitions:
Block ge reaching definitions:
  From block 0 : var cond0 instr 1
  From block 0 : var b instr 0
Block zero reaching definitions:
  From block 0 : var cond0 instr 1
  From block 0 : var b instr 0
Block end reaching definitions:
  From block 0 : var cond0 instr 1
  From block ge : var b instr 1
  From block zero : var b instr 1

